spectree/plugins/quart_plugin.py

```diff
@@ -0,0 +1,216 @@
+                    not str(rule).startswith(self.blueprint_state.url_prefix)
+                    or str(rule).startswith(
+                        "/".join([self.blueprint_state.url_prefix, self.config.path])
+                    )
+                )
+            ):
+                continue
+            yield rule
+
+    def parse_func(self, route: Any):
+        from quart import current_app
+
+        if self.blueprint_state:
+            func = self.blueprint_state.app.view_functions[route.endpoint]
+        else:
+            func = current_app.view_functions[route.endpoint]
+
+        # view class: https://flask.palletsprojects.com/en/1.1.x/views/
+        if getattr(func, "view_class", None):
+            cls = getattr(func, "view_class")
+            for method in route.methods:
+                view = getattr(cls, method.lower(), None)
+                if view:
+                    yield method, view
+        else:
+            for method in route.methods:
+                yield method, func
+
+    def _fill_form(self, request) -> dict:
+        req_data = get_multidict_items(request.form)
+        req_data.update(get_multidict_items(request.files) if request.files else {})
+        return req_data
+
+    async def request_validation(self, request, query, json, form, headers, cookies):
+        """
+        req_query: werkzeug.datastructures.ImmutableMultiDict
+        req_json: dict
+        req_headers: werkzeug.datastructures.EnvironHeaders
+        req_cookies: werkzeug.datastructures.ImmutableMultiDict
+        """
+        req_query = get_multidict_items(request.args)
+        req_headers = dict(iter(request.headers)) or {}
+        req_cookies = get_multidict_items(request.cookies) or {}
+        has_data = request.method not in ("GET", "DELETE")
+        use_json = json and has_data and request.mimetype == "application/json"
+        use_form = (
+            form
+            and has_data
+            and any([x in request.mimetype for x in self.FORM_MIMETYPE])
+        )
+
+        request.context = Context(
+            query.parse_obj(req_query) if query else None,
+            json.parse_obj(await request.get_json(silent=True) or {})
+            if use_json
+            else None,
+            form.parse_obj(self._fill_form(request)) if use_form else None,
+            headers.parse_obj(req_headers) if headers else None,
+            cookies.parse_obj(req_cookies) if cookies else None,
+        )
+
+    async def validate(
+        self,
+        func: Callable,
+        query: Optional[ModelType],
+        json: Optional[ModelType],
+        form: Optional[ModelType],
+        headers: Optional[ModelType],
+        cookies: Optional[ModelType],
+        resp: Optional[Response],
+        before: Callable,
+        after: Callable,
+        validation_error_status: int,
+        skip_validation: bool,
+        *args: Any,
+        **kwargs: Any,
+    ):
+        from quart import abort, jsonify, make_response, request
```

Comment:
I see your point. Let me check if we can avoid this.

Reply:
fixed